receiver.go

```diff
@@ -108,39 +52,113 @@ func (r *Receiver) Receive(ctx context.Context) (*Message, error) {
 		}
 	}
 
+	// for ModeFirst, auto-accept the message
+	modeFirstAccept := func(ctx context.Context, msg *Message) (*Message, error) {
+		if r.manualCreditor != nil || receiverSettleModeValue(r.link.ReceiverSettleMode) == ModeSecond {
```

Comment:
I don't think we should bypass modeFirstAccept for the manual creditor if I'm getting it (it's the equivalent of 'receive and delete', right?). 

If so, auto-accept is still correct in that mode, manual or not.

Reply:
I see, I misunderstood the comments in the other thread.  Will remove.

Reply:
Correct mode first is receive and delete. You ack the receive before processing it.